src/main/java/com/github/junrar/Junrar.java

```diff
@@ -90,6 +90,24 @@
         return contents;
     }
 
+    public static List<ContentDescription> getContentsDescription(final InputStream resourceAsStream) throws RarException, IOException {
+        final Archive arch = createArchiveOrThrowException(resourceAsStream, null);
+
+        final List<ContentDescription> contents = new ArrayList<>();
+        try {
+            if (arch.isEncrypted()) {
+                logger.warn("archive is encrypted cannot extract");
+                return new ArrayList<>();
+            }
+            for (final FileHeader fileHeader : arch) {
+                contents.add(new ContentDescription(fileHeader.getFileName(), fileHeader.getUnpSize()));
+            }
+        } finally {
+            arch.close();
+        }
+        return contents;
```

Comment:
You can probably pull that duplicate code out in a private function

Reply:
Absolutely !